addons/api/addon/transforms/object.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Transform from '@ember-data/serializer/transform';
+import { TrackedObject } from 'tracked-built-ins';
+import { typeOf } from '@ember/utils';
+
+export default class ObjectTransform extends Transform {
+  /**
+   * @param serialized
+   * @returns {object}
+   */
+  deserialize(serialized) {
+    if (typeOf(serialized) === 'object') {
+      const obj = new TrackedObject(serialized);
+      return obj;
+    } else {
+      return {};
+    }
+  }
+
+  /**
+   * @param deserialized
+   * @returns {object}
+   */
+  serialize(deserialized) {
+    if (typeOf(deserialized) === 'object') {
+      const obj = new TrackedObject(deserialized);
```

Comment:
I don't think we need to turn this into a tracked object. I think we should be serializing it back into just a normal object, though it might just work fine even without doing that